### ISSUE TITLE
Flaky test: FrameConnectionManagerTests.UnrootedConnectionsGetRemovedFromHeartbeat

### ISSUE THREAD

**Issue by Tratcher** [aspnet/KestrelHttpServer#1725]
```
Assert.Equal() Failure
Expected: 1
Actual:   0
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.FrameConnectionManagerTests.UnrootedConnectionsGetRemovedFromHeartbeatInnerScope(String connectionId, FrameConnectionManager frameConnectionManager, Mock`1 trace) in C:\BuildAgent\work\2b0c8ce712895332\.r\KestrelHttpServer\test\Microsoft.AspNetCore.Server.Kestrel.Core.Tests\FrameConnectionManagerTests.cs:line 58
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.FrameConnectionManagerTests.UnrootedConnectionsGetRemovedFromHeartbeat() in C:\BuildAgent\work\2b0c8ce712895332\.r\KestrelHttpServer\test\Microsoft.AspNetCore.Server.Kestrel.Core.Tests\FrameConnectionManagerTests.cs:line 26
```

**Comment by Tratcher** [comment]
http://aspnetci/viewLog.html?buildId=205441&tab=buildResultsDiv&buildTypeId=XPlat_Windows_Win10Universe

**Comment by davidfowl** [comment]
Also happened on appveyor with @halter73 's initial commit

**Comment by davidfowl** [comment]
It happened again on @natemcmaster 's PR https://ci.appveyor.com/project/aspnetci/kestrelhttpserver/build/1.0.5088. I skipped the test https://github.com/aspnet/KestrelHttpServer/commit/d0743806d8d8675e82afd665ecdd9bd208564cf7

**Comment by halter73** [comment]
Fixed by #1727.
